src/modules/sequencer-crosshair/CrosshairsPlaceable.js

```diff
@@ -0,0 +1,319 @@
+						this.crosshair.location.offsetDistance * this.document.parent.grid.size
+					).B;
+
+					const snappedPosition = this.getSnappedPoint(edgeLocation);
+
+					const { rotation, distance } = this._getDraggedMatrix(snappedPosition, mouseLocation);
+
+					this.document.updateSource({
+						x: snappedPosition.x,
+						y: snappedPosition.y,
+						distance,
+						direction: rotation
+					});
+				}
+			} else if (this.crosshair.location.limit) {
+
+				const ray = new Ray({ x: locationX, y: locationY }, mouseLocation);
+				const rayDistance = ray.distance / this.document.parent.grid.size;
+
+				const min = this.crosshair.location.minDistance ?? 0;
+				const max = this.crosshair.location.maxDistance ?? Infinity;
+				const cappedDistance = Math.max(min, Math.min(max, rayDistance));
+
+				const snappedPosition = this.getSnappedPoint(ray.project(cappedDistance / rayDistance));
+
+				const { rotation, distance } = this._getDraggedMatrix(snappedPosition, mouseLocation);
+
+				this.document.updateSource({
+					x: snappedPosition.x,
+					y: snappedPosition.y,
+					distance,
+					direction: rotation
+				});
+
+			}
+
+		} else if (this.#isDrag) {
+			const { rotation, distance } = this._getDraggedMatrix(this.document, mouseLocation);
+			this.document.updateSource({
+				distance,
+				direction: rotation
+			});
+		} else if (!this.crosshair.location.obj) {
+			const { x, y } = this.getSnappedPoint(mouseLocation);
+			this.document.updateSource({ x, y });
+		}
+
+	}
+
+	_getDraggedMatrix(source, target, snapDirection = this.crosshair.snap.direction) {
+
+		const drag = new Ray(source, target);
+
+		let distance = Math.max(0.5, drag.distance / this.document.parent.grid.size * this.document.parent.grid.distance);
+
+		if (this.crosshair.distanceMinMax.min && this.crosshair.distanceMinMax.max) {
+			distance = Math.min(Math.max(distance, this.crosshair.distanceMinMax.min), this.crosshair.distanceMinMax.max);
+		}
+
+		const rotation = snapDirection
+			? Math.round(Math.toDegrees(drag.angle) / snapDirection) * snapDirection
+			: Math.toDegrees(drag.angle);
+
+		return { rotation, distance };
+
+	}
+
+	/** @override */
+	_destroy(options = {}) {
+		super._destroy(options);
+		this._clearHandlers();
+	}
+
+	_clearHandlers(evt) {
+		this.layer.interactiveChildren = this.oldInteractiveChildren;
+		canvas.stage.off("mousemove", this.#handlers.move);
+		canvas.stage.off("mouseup", this.#handlers.confirm);
+		canvas.app.view.oncontextmenu = null;
+		canvas.app.view.onwheel = null;
+		canvas.mouseInteractionManager.reset({ interactionData: true, state: false });
+	}
+
+	_onConfirm(evt) {
+		evt.preventDefault();
+		canvas.mouseInteractionManager.cancel(evt);
+		if (this.#isDrag) {
+			this.#isDrag = false;
+			canvas.mouseInteractionManager.reset({
+				interactionData: true,
+				state: false,
+			});
+			return;
+		}
+
+		this.document.endPosition = this.document.t === CONST.MEASURED_TEMPLATE_TYPES.CONE || CONST.MEASURED_TEMPLATE_TYPES.RAY

```

Comment:
Was this meant to be `(this.document.t === CONST.MEASURED_TEMPLATE_TYPES.CONE) || (this.document.t === CONST.MEASURED_TEMPLATE_TYPES.RAY)`? Currently `CONST.MEASURED_TEMPLATE_TYPES.RAY` is being cast to a boolean which I don't think is the intent.